.github/workflows/ci.yml

```
@@ -0,0 +1,40 @@
+name: Continuous Integration
+
+on:
+  pull_request:
+    branches: [master]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '22.13.1'
+      - name: Install Dependencies
+        run: npm install
+      - name: Run Build Check
+        run: npm run build
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '22.13.1'
+      - name: Install Dependencies
+        run: npm install
+      - name: Run Linter Checks
+        run: npm run lint
+  prettier:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '22.13.1'
+      - name: Install Dependencies
+        run: npm install
+      - name: Run Prettier Checks
+        run: npm run prettify:check
```

I'd only one job, you are creating a job per task, but it should be a job with multiple steps. This will be mor mantenible and you can save time (for example now you are installing the dependencies multiple times)

```suggestion
jobs:
  checks:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: '22.13.1'
      - name: Install Dependencies
        run: npm install
      - name: Run Linter Checks
        run: npm run lint
      - name: Run Prettier Checks
        run: npm run prettify:check
      - name: Run Build Check
        run: npm run build
```